R/bootstrap.nlsfit.R

```diff
@@ -1247,23 +1247,11 @@ plot.bootstrapfit <- function(x, ..., col.line="black", col.band="gray", opacity
   my.xlim <- mylims$xlim
   my.ylim <- mylims$ylim
 
-  ## to include additional parameter to x$fn originally given as ... to
-  ## bootstrap.nlsfit requires some pull-ups
-  Y <- do.call(x$fn, c(list(par = x$t0[1:npar], x = X, boot.r = 0), x$tofn))
-
-  if(!is.null(x$t)) {
-    ## error band
-    ## define a dummy function to be used in apply
-    prediction_boot_fn <- function (boot.r) {
-      par <- x$t[boot.r, 1:npar, drop = FALSE]
-      do.call(x$fn, c(list(par = par, x = X, boot.r = boot.r), x$tofn))
-    }
-    predictions <- do.call(rbind, lapply(1:nrow(x$t), prediction_boot_fn))
-    se <- apply(predictions, 2, error, na.rm = TRUE)
-    stopifnot(length(se) == length(X))
+  prediction <- predict(x, X)
 
-    ## plot it
-    polyval <- c(Y+se, rev(Y-se))
+  ## error band
+  if(!is.null(prediction$err)) {
+    polyval <- c(prediction$val + prediction$err, rev(prediction$val - prediction$err))
     if(any(polyval < my.ylim[1]) || any(polyval > my.ylim[2])) {
       polyval[polyval < my.ylim[1]] <- my.ylim[1]
```

Comment:
I was always wondering why the error bands of bootstrap.nlsfit looked so crappy... why do you restrict them to the plot region?

Reply:
You mean that they should also extrapolate a bit?